STYLEGUIDE.md

```diff
@@ -0,0 +1,101 @@
+- (Not) **Recommended** and **Preferred** clauses should be followed where practicable; a brief rationale should be provided for not doing so
+- **Suggested** clauses indicate a convention, approach or guideline that may be helpful to follow, but has no binding effect or justification requirements
+- **May** or **Can** clauses designate explicitly permitted guidelines, actions or conventions, often for clarity, that are allowed but have no weight of preference on them (and may even be gently discouraged with a suggested or preferred alternative)
+
+
+## Structure
+
+- In general, follow the [OpenStack Heading Guidelines](https://docs.openstack.org/doc-contrib-guide/writing-style/general-writing-guidelines.html) with regard to sections and headings; in particular, section and subsection titles delineating overall categories of tasks/topics and be in gerund (i.e. with "ing", e.g. "Editing code", "Debugging files"), while those concerning a concept or reference topic should be a noun (e.g. "General considerations", "Requirements") and lowest level topic/task titles should generally be in imperative ("Create a code cell", "Get help", etc) where reasonable
+- Heading case: Page titles (L1 headings) should use Title Case, all others should use Sentence case
+- 1 L1 Heading for page titles: Use ``######`` with overline; on first line of file; keep to a few words
+- L2 Headings for section heads: Use ========= with overline; 3 blank lines before them, will appear in sidebar so keep them short
+- L3 headings for subheads: Use ~~~~~~, 2 blank lines,
+- L4 Headings for sub-subheads if truly necessary: Use ---------, 2 blank lines, keep to a minimum (consider a flatter structure or breaking into multiple files)
+- Related components at the bottom: L3 heading, bullets, 2 blank lines before, try for 2-4 links if included
+- All files with Related components should contain at least one L2 section; if only one, can be  entitled Overview
+- Typically, for files with multiple sections or longer than one page, a summary section of a few lines length is helpful before the first section, summarizing the functionality discussed in the section and its purpose
+- Name of component/topic should be in first sentence, Title Case, bold on first use
+
+
+## Technical
+
+- Markdown, Github-flavored (``.md``) for any non-documentation rendered text files in the root of the repo, e.g. ``README``, ``CONTRIBUTING``, etc.
+- reStructuredText, (reST, ``.rst``) for documentation format, i.e. any user-visible files in the ``/doc`` directory
+- Where not otherwise contradicted by this guide or specific individual requirements, a subset of Markdown corresponding to that cross-compatible with reST should be preferred where possible; e.g. ``*italic*`` vs. ``_italic_``, ``*`` for bulleted lists, double backtick for inline code vs. single, etc.
```

Comment:
I've addressed this before, at some length. For Markdown in the Spyder repo it may be justifiable since it is less likely to have to interoperate with the reST docs, Markdown is the common format, you have to edit it frequently and edits are much more short-form. However, given none of this is the case (and noting that on an US English keyboard layout, which would be expected to be used/switched to when writing the long-form English prose on this repo, the additional time spent is minuscule relative to the substantial overhead converting back and forth if needed, even with tools), I judged it to be worthwhile and the content of this repo is all formatted accordingly.

If you and @juanis2112 disagree and want to change it, that's fine, but the style guide is meant to reflect the current state of the docs and repo, where this is the case. Once this is committed, if you want to change it, it can then be changed repo-wide and in the style guide, and the changes submitted as an atomic PR, so its clear.